src/Components/CreateImageWizardV2/EditImageWizard.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect } from 'react';
+
+import { useNavigate } from 'react-router-dom';
+
+import CreateImageWizard from './CreateImageWizard';
+import { mapRequestToState } from './utilities/requestMapper';
+
+import { useAppDispatch } from '../../store/hooks';
+import { useGetBlueprintQuery } from '../../store/imageBuilderApi';
+import { loadWizardState, initializeWizard } from '../../store/wizardSlice';
+import { resolveRelPath } from '../../Utilities/path';
+
+type EditImageWizardProps = {
+  composeId: string;
+};
+
+const EditImageWizard = ({ composeId }: EditImageWizardProps) => {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const { data: blueprintDetails, error } = useGetBlueprintQuery({
+    id: composeId,
+  });
+  useEffect(() => {
+    if (composeId && blueprintDetails) {
+      const editBlueprintState = mapRequestToState(blueprintDetails);
+      dispatch(loadWizardState(editBlueprintState));
+    } else {
+      dispatch(initializeWizard());
+    }
+  }, [composeId, blueprintDetails, dispatch]);
+  useEffect(() => {
+    // redirect to the main page if the composeId is invalid
+    if (error) navigate(resolveRelPath(''));
+  }, [error, navigate]);
```

Comment:
Possibly beyond the scope of this PR, but would like to have the conversation here and create an issue to track it if you agree...

Do you think it would be a good idea to issue a notification here letting the user know that we couldn't resolve the `blueprintId`? Right now we fail silently which might be confusing/frustrating.